pkg/controller/controller.go

```diff
@@ -203,6 +232,28 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	capacity := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	volSizeBytes := capacity.Value()
 
+	userCredentials := map[string]string{}
+	storageClassName := options.PVC.Spec.StorageClassName
+	fmt.Printf("><SB> provisioner was called for storage class: %s\n", *storageClassName)
+	storageClass, err := p.client.StorageV1().StorageClasses().Get(*storageClassName, metav1.GetOptions{})
+	if err != nil {
+		glog.Warningf("failed to retrieve information about the storage class: %s with error: %v", storageClassName, err)
+	} else {
```

Comment:
Uneccessary else-block.

Reply:
Appreciate if you could clarify why. The logic here is if we cannot get a storage class object, we just print warning and do not process any secret info, basically just skipping whatever is covered by else part.

Reply:
It looks like storageclass is optional.  If so, then the code is good.